meteor/sofie-server/data-access/repositories/test/mongo-test-database.ts

```diff
@@ -30,6 +30,35 @@ export class MongoTestDatabase {
 		}
 	}
 
+	public getValidObjectIdString(base: string): string {
+		const twelveChar = base.length >= 12 ? base.substring(0, 12) : this.fillTo12Chars(base).substring(0, 12)
+		return this.convertToHex(twelveChar)
+	}
+
+	private fillTo12Chars(base: string): string {
+		const missingLength = 12 - base.length + 2
+		const multiplier = this.replaceAll('1' + Array<number>(missingLength).fill(0).join(), ',')
+		return base + Math.floor(Math.random() * +multiplier)
+	}
+
+	// Can and should be replaced with 'string.replaceAll(...)', when the project is updated to target Es2021.
+	private replaceAll(target: string, searchFor: string, replaceWith: string = ''): string {
+		while (target.includes(searchFor)) {
+			target = target.replace(searchFor, replaceWith)
+		}
+		return target
+	}
+
+	private convertToHex(base: string): string {
+		let hex = ''
+		for (let i = 0; i < base.length; i++) {
+			const charCode = base.charCodeAt(i)
+			const hexValue = charCode.toString(16)
+			hex += hexValue.padStart(2, '0')
+		}
+		return hex
+	}
+
```

Comment:
This seems way overcomplicated. What is the reasoning for this?

Reply:
In the test files where you need a valid id for a mongo entity you could just use `ObjectId().toString()`.

Another suggestion is that we abandon the use of `_id`, such that it is only Mongo that uses it under the hood. Instead we could add our own attribute, e.g. `id`, which will act as the identifier for the entities. But this is most likely something we should do when sofie-server has more control over the write access to the database.

Reply:
The commented on code has been removed as of https://github.com/tv2/tv-automation-server-core/pull/255/commits/a7a6af9a42a96f6b8e1507e3d9969c0ee6317fd0

Thanks to input from both of you.